src/test/mir-opt/const_prop/dodge_bad_ops_that_never_happen.rs

```diff
@@ -0,0 +1,25 @@
+// This test is designed to never trigger `unconditional_panic`.
+// If it does, there's a bug in the MIR optimizations.
+#![deny(unconditional_panic)]
+// EMIT_MIR rustc.main.ConstProp.diff
```

Comment:
you're emitting the wrong function's MIR for this test I think ;)

Reply:
The directive emits _per function_? 😆 I didn't know! Okay, I'll put it where it is needed then :D

Reply:
ahh, `rustc.`**`main`**`.ConstProp.diff` :D

Reply:
Now that I think of it... should we be emitting the MIR for this test? Since the MIR generated isn't quite the point of this test.

I mean I generated it out of habit, but if we don't need it we may as well just delete it right? To ease in maintenance burden.

Reply:
I thought the point of the test was to look at the MIR of `combine_wisely` so we have a sort of canary for the MIR changes that can happen here? If you're just worried about new diagnostics popping up, you should make it a `ui` test

Reply:
Nono, the point of this test is that it should never fail to compile. If it does, it's extremely likely that it happened because we messed up const-prop and triggered an `unavoidable_panic` lint.

Reply:
re: ui test.
Wait. Can I make a ui test that has no output? Because ideally this would have no compiler output whatsoever. I only know the ui tests that are like "error outputs" or "warning outputs", maybe execution outputs... that sort of stuff. Can we have a ui test that's just like "yep, this should give no special output"?

Reply:
you can make a `// check-pass` ui test that just is supposed to compile successfully.

Reply:
Ahh, then that's what it should be really. Done! Thanks Oli <3